melodies_monet/driver.py

```diff
@@ -1326,6 +1339,12 @@ def pair_data(self, time_interval=None):
                 # TODO: add other network types / data types where (ie flight, satellite etc)
                 # if sat_swath_clm (satellite l2 column products)
                 elif obs.obs_type.lower() == 'sat_swath_clm':
+                    # grab kwargs for pairing. Use default if not specified
+                    if obs.obs_type.lower() in list(self.pairing_kwargs.keys()):
+                        pairing_kws = self.pairing_kwargs[obs.obs_type.lower()]
+                    else: 
+                        print('WARNING: The satellite pairing option apply_ak is being set to True. Pairing will fail if there is no AK available.')
+                        pairing_kws = {'apply_ak': True, 'mod_to_overpass': False}
```

Comment:
Currently you'd have an issue later if `pairing_kwargs` for sat were set in the control but didn't contain `apply_ak`, e.g. just `mod_to_overpass` or another unrelated setting. An alternative approach would be to set `pairing_kws` to your desired defaults first and then update it based on `pairing_kwargs` from the control. That way you know everything is set. There would still be an opportunity to let the user know that they may need to set `apply_ak`.